### DISCUSSION TITLE
Reusable matrix blocks

### DISCUSSION THREAD

**Discussion by scoch5** [craftcms/cms#14601] [Ideas] [answered]
Since the matrix blocks are now post, the possibility of adding and modifying a previously created block could be added. Kind of like Gutenberg's reusable blocks.

**Comment by brandonkelly** [comment]
Not sure I’m following. Can you elaborate?

**Reply by scoch5** [reply]
You could give the possibility to add a block (entry) already created.
Let me explain better: let's say we have a Matrix field with different EntryTypes to define the content of the page and we would like one of these blocks to have the same content as one present in another page. For now we can only fill in the fields with the same values without these two blocks being actually connected.
It would be nice to be able to insert already created blocks onto other pages so that their contents are truly connected.

**Reply by thupsi** [reply]
You can do that if you have a section with entries that act as blocks. You then can use them via an entry field.

**Comment by brandonkelly** [comment, marked as answer]
I don’t think Matrix fields should mix nested entries with entries that are shared by other Matrix fields.

Instead, you could create a new section that contains the shared content, and then use an Entries field within the Matrix field to relate those shared entries (as @thupsi suggested).

**Reply by scoch5** [reply]
I understand what you're saying but I think having the functionality natively can greatly improve the editing experience.

**Reply by thupsi** [reply]
Well, it is native, isn't it? I mean the section with the blocks could be called "template partials" , or "reusable blocks". And you could even use those as defaults and _partially_ override their values with the nested matrix field values. Am I missing something?

**Reply by thupsi** [reply]
One possible missing convenience that comes to mind would be the ability to copy/clone an existing block (entry) in place, so you could use that as a starting point. @brandonkelly Do you think it would make sense to add something along those lines?

**Reply by thupsi** [reply]
I'm rambling a bit, but, to clarify, this duplication is already possible within a "reusable blocks" section. I was talking about being able to copy an entry _from_ that section to a matrix field. Maybe hook into/extend [this functionality](https://github.com/craftcms/cms/pull/14541)?
